### PR TITLE
Use NftID instead of UniqueTokenId

### LINKED ISSUE #7346: Use NftID instead of UniqueTokenId
1. Move NftID from token_get_nft_info.proto to basic_types.proto
2. Change everywhere that uses UniqueTokenID to use NftID instead
3. Delete UniqueTokenId from common.proto
4. Update all dependent code

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
In [PR #292](https://github.com/hashgraph/hedera-protobufs/pull/292) over in `hedera-protobufs`, made the following changes:
1) Move `NftID` from `token_get_nft_info.proto` to `basic_types.proto`
2) Change everywhere that uses `UniqueTokenID` to use `NftID` instead
3) Delete `UniqueTokenId` from `common.proto`

In this PR, we update all services code that depends on the above.
 
**Related issue(s)**:
Fixes #7346 

**Notes for reviewer**:
This PR relies on a temporary branch `7346-change-UniqueTokenId-to-NftID` in `hedera-protobufs` repo. In order to avoid breaking develop in between the time that changes are made in `hedera-protobufs` and `hedera-services`, the plan is to:

1) Create temporary branch `7346-change-UniqueTokenId-to-NftID` in `hedera-protobufs` that contains the changes listed above. This is done. It is the branch associated with PR #[292](https://github.com/hashgraph/hedera-protobufs/pull/292)
2) Merge this PR
3) Merge #[292](https://github.com/hashgraph/hedera-protobufs/pull/292)  in hedera-protobufs repo, but DO NOT DELETE THE ASSOCIATED BRANCH YET. Until Step 4 is completed, `hedera-services` still relies on the branch
4) Merge PR #7410 to reset the protobufs branch in use back to `add-pbj-types-for-state`
5) Delete the `hedera-protobufs` branch associated with steps 1 and 3

### CODE REVIEW

File: settings.gradle.kts
Comment (mhess-swl) [inline settings.gradle.kts:106]: intentionally kept?

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java:62]: typo in method name `merkelUniqueTokenToNftID ` to `merkleUniqueTokenToNftID`

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java:67]: Probably rename to `nftNumPairToNftID`

**Review by Neeharika-Sompalli — COMMENTED** [review]
Have two comments on naming , otherwise LGTM!  But seems some unit tests are failing in CI

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java
Comment (kimbor) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java]: Fixed

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java
Comment (kimbor) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/NftStateTranslator.java]: Fixed

**Review by mhess-swl — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Comment by JeffreyDallas** [comment]
I would suggest to run one or two test panel and post test result here before the merge

https://github.com/hashgraph/hedera-services/actions/workflows/node-flow-fsts-custom-regression.yaml

**Comment by Neeharika-Sompalli** [comment]
> I would suggest to run one or two test panel and post test result here before the merge
> 
> https://github.com/hashgraph/hedera-services/actions/workflows/node-flow-fsts-custom-regression.yaml

@JeffreyDallas  This should not cause any problems wih JRS tests as the changes are in modularized codebase.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! @kimbor Thanks ! Please make sure to not break while merging protobuf changes.

**Comment by JeffreyDallas** [comment]
> > I would suggest to run one or two test panel and post test result here before the merge
> > https://github.com/hashgraph/hedera-services/actions/workflows/node-flow-fsts-custom-regression.yaml
> 
> @JeffreyDallas This should not cause any problems wih JRS tests as the changes are in modularized codebase.

I hope so, but I did see some failures already, will post the summary result here once they are done

**Review by JeffreyDallas — CHANGES_REQUESTED** [review]
Please wait regression results before merge

**Review by rbair23 — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**
